{"uid":"d6caae5cc969b65e","name":"Если кликнуть на заказ, откроется всплывающее окно с деталями","fullName":"tests.test_orders_feed.TestOrdersFeed#test_click_by_order_opens_the_order_details","historyId":"ad7b3374536ee8dff4ccd000724a1edd","time":{"start":1707757934060,"stop":1707757934082,"duration":22},"description":"Если кликнуть на заказ, откроется всплывающее окно с деталями","descriptionHtml":"<p>Если кликнуть на заказ, откроется всплывающее окно с деталями</p>\n","status":"broken","statusMessage":"selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: .//a[contains(@class,\"OrderHistory_link\")]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:189:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:507:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:132:16","statusTrace":"self = <tests.test_orders_feed.TestOrdersFeed object at 0x10216e8e0>\nfeed_page = <pages.feed_page.FeedPage object at 0x102133670>\n\n    @allure.title('Если кликнуть на заказ, откроется всплывающее окно с деталями')\n    @allure.description('Если кликнуть на заказ, откроется всплывающее окно с деталями')\n    def test_click_by_order_opens_the_order_details(self, feed_page):\n>       feed_page.get_first_order_link().click()\n\ntests/test_orders_feed.py:17: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/feed_page.py:22: in get_first_order_link\n    return self.find_element(FeedPageLocators.LINK_FIRST_ORDER)\npages/base_page.py:23: in find_element\n    element = self.driver.find_element(*locator)\n../../../arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py:741: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n../../../arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x102133970>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"Unable to locate element: .//a[contains(@clas.../content/shared/webdriver/Errors.sys.mjs:507:5\\\\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:132:16\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: .//a[contains(@class,\"OrderHistory_link\")]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nE       WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:189:5\nE       NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:507:5\nE       dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:132:16\n\n../../../arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py:229: NoSuchElementException","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"driver","time":{"start":1707757930758,"stop":1707757932440,"duration":1682},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"attachmentStep":false},{"name":"feed_page","time":{"start":1707757932440,"stop":1707757934059,"duration":1619},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"attachmentStep":false}],"testStage":{"description":"Если кликнуть на заказ, откроется всплывающее окно с деталями","status":"broken","statusMessage":"selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: .//a[contains(@class,\"OrderHistory_link\")]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:189:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:507:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:132:16","statusTrace":"self = <tests.test_orders_feed.TestOrdersFeed object at 0x10216e8e0>\nfeed_page = <pages.feed_page.FeedPage object at 0x102133670>\n\n    @allure.title('Если кликнуть на заказ, откроется всплывающее окно с деталями')\n    @allure.description('Если кликнуть на заказ, откроется всплывающее окно с деталями')\n    def test_click_by_order_opens_the_order_details(self, feed_page):\n>       feed_page.get_first_order_link().click()\n\ntests/test_orders_feed.py:17: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/feed_page.py:22: in get_first_order_link\n    return self.find_element(FeedPageLocators.LINK_FIRST_ORDER)\npages/base_page.py:23: in find_element\n    element = self.driver.find_element(*locator)\n../../../arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py:741: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n../../../arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x102133970>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"Unable to locate element: .//a[contains(@clas.../content/shared/webdriver/Errors.sys.mjs:507:5\\\\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:132:16\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: .//a[contains(@class,\"OrderHistory_link\")]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nE       WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:189:5\nE       NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:507:5\nE       dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:132:16\n\n../../../arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py:229: NoSuchElementException","steps":[{"name":"Найти ссылку на первый заказ","time":{"start":1707757934060,"stop":1707757934080,"duration":20},"status":"broken","statusMessage":"selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: .//a[contains(@class,\"OrderHistory_link\")]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:189:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:507:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:132:16\n\n","statusTrace":"  File \"/Users/lkitova/arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/allure_commons/_allure.py\", line 173, in impl\n    return func(*a, **kw)\n  File \"/Users/lkitova/Documents/Diplom/Diplom_3/pages/feed_page.py\", line 22, in get_first_order_link\n    return self.find_element(FeedPageLocators.LINK_FIRST_ORDER)\n  File \"/Users/lkitova/arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/allure_commons/_allure.py\", line 173, in impl\n    return func(*a, **kw)\n  File \"/Users/lkitova/Documents/Diplom/Diplom_3/pages/base_page.py\", line 23, in find_element\n    element = self.driver.find_element(*locator)\n  File \"/Users/lkitova/arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py\", line 741, in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n  File \"/Users/lkitova/arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/Users/lkitova/arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n","steps":[{"name":"Ищем элемент на странице","time":{"start":1707757934060,"stop":1707757934071,"duration":11},"status":"broken","statusMessage":"selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: .//a[contains(@class,\"OrderHistory_link\")]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:189:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:507:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:132:16\n\n","statusTrace":"  File \"/Users/lkitova/arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/allure_commons/_allure.py\", line 173, in impl\n    return func(*a, **kw)\n  File \"/Users/lkitova/Documents/Diplom/Diplom_3/pages/base_page.py\", line 23, in find_element\n    element = self.driver.find_element(*locator)\n  File \"/Users/lkitova/arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py\", line 741, in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n  File \"/Users/lkitova/arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/Users/lkitova/arcadia/games/qa/qa_tests/venv/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n","steps":[],"attachments":[],"parameters":[{"name":"locator","value":"('xpath', './/a[contains(@class,\"OrderHistory_link\")]')"}],"stepsCount":0,"hasContent":true,"shouldDisplayMessage":true,"attachmentsCount":0,"attachmentStep":false}],"attachments":[],"parameters":[],"stepsCount":1,"hasContent":true,"shouldDisplayMessage":false,"attachmentsCount":0,"attachmentStep":false}],"attachments":[{"uid":"eef7fd08154596c3","name":"stderr","source":"eef7fd08154596c3.txt","type":"text/plain","size":131}],"parameters":[],"stepsCount":2,"hasContent":true,"shouldDisplayMessage":true,"attachmentsCount":1,"attachmentStep":false},"afterStages":[{"name":"driver::0","time":{"start":1707757934106,"stop":1707757934803,"duration":697},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"attachmentStep":false}],"labels":[{"name":"parentSuite","value":"tests"},{"name":"suite","value":"test_orders_feed"},{"name":"subSuite","value":"TestOrdersFeed"},{"name":"host","value":"i111355119"},{"name":"thread","value":"32733-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.test_orders_feed"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"driver","value":"'firefox'"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"tags":[]},"source":"d6caae5cc969b65e.json","parameterValues":["'firefox'"]}